ruby-service/app/circuit_breaker.rb

```
@@ -0,0 +1,392 @@
+
+    def record_rejection
+      @mutex.synchronize { @rejected_calls += 1 }
+    end
+
+    def record_state_transition
+      @mutex.synchronize { @state_transitions += 1 }
+    end
+
+    def average_response_time
+      @mutex.synchronize do
+        return 0 if @response_times.empty?
```

RuboCop (1.88.2)

[convention] Add empty line after guard clause.

(Layout/EmptyLineAfterGuardClause)